src/blocks/mrc_component.ts

```diff
@@ -25,18 +25,43 @@ import { Order } from 'blockly/python';
 import { MRC_STYLE_COMPONENTS } from '../themes/styles'
 import { createFieldNonEditableText } from '../fields/FieldNonEditableText';
 import { ExtendedPythonGenerator } from '../editor/extended_python_generator';
-import { getAllowedTypesForSetCheck } from './utils/python';
+import { getAllowedTypesForSetCheck, getClassData, getSubclassNames } from './utils/python';
+import * as ToolboxItems from '../toolbox/items';
+import * as CommonStorage from '../storage/common_storage';
+import { createPortShadow } from './mrc_port';
+import { createNumberShadowValue } from './utils/value';
 
 export const BLOCK_NAME = 'mrc_component';
 export const OUTPUT_NAME = 'mrc_component';
 
+export const FIELD_NAME = 'NAME';
+export const FIELD_TYPE = 'TYPE';
+
+// key is a regex pattern that matches a function argument name, value is type of mrc_port block to use.
+// TODO: Improve these regex pattern.
+// TODO: Should the types here match the PortType values in external_samples/component.py?
+const RECOGNIZED_PORT_ARG_NAME_PATTERNS: { [key: string]: string } = {
+  'can_port': 'SmartCan',
+  'i2c_port': 'SmartIO',
```

Comment:
I2C is its own type of port, it shouldn't be SmartIO

Reply:
Done.